syntax = "proto3";

package caf.rpc;

service App {
  rpc Quit(Null) returns (Null) {}
  rpc GetAppInfo(Null) returns (AppInfoReply) {}
}

message Null {}

enum AppType
{
  CONSOLE     = 0x00;
  GRPC_SERVER = 0x01;
  GRPC_CLIENT = 0x02;
  GUI         = 0x04;
  WEB         = 0x08;
}

message AppInfoReply
{
  string name         = 1;
	uint32 type         = 2;
  int32 major_version = 3;
  int32 minor_version = 4;
  int32 patch_version = 5;
}

